transformations/close_homophones_swap/transformation.py

```diff
@@ -0,0 +1,59 @@
+import itertools
+import random
+import spacy
+import numpy
+
+from interfaces.SentenceOperation import SentenceOperation
+from tasks.TaskTypes import TaskType
+from SoundsLike.SoundsLike import Search
+from spacy.attrs import LOWER, POS, ENT_TYPE, IS_ALPHA
+from spacy.tokens import Doc
+
+"""
+Base Class for implementing the different input transformations a generation should be robust against.
+"""
+
+
+def close_homophones_swap(text, corrupt_prob, seed=0, max_output=1):
+    random.seed(seed)
+    nlp = spacy.load("en_core_web_sm")
```

Comment:
Hi @KaizhaoLiang thank you for the changes!

It might be better if spacy is loaded once (in the constructor).

Reply:
Updated, thanks for the suggestion